coordinator/persistence-api/src/main/java/org/apache/cassandra/stargate/transport/ProtocolVersion.java

```diff
@@ -55,40 +57,50 @@ public enum ProtocolVersion implements Comparable<ProtocolVersion> {
     this.beta = beta;
   }
 
+  /** Some utility constants for decoding DSE versions */
+  private static final byte DSE_VERSION_BIT = 0x40; // 0100 0000
+
   /**
    * The supported versions stored as an array, these should be private and are required for fast
    * decoding
    */
-  private static final ProtocolVersion[] SUPPORTED_VERSIONS = new ProtocolVersion[] {V3, V4, V5};
+  private static final ProtocolVersion[] OS_VERSIONS = new ProtocolVersion[] {V3, V4, V5};
+
+  static final ProtocolVersion MIN_OS_VERSION = OS_VERSIONS[0];
+  static final ProtocolVersion MAX_OS_VERSION = OS_VERSIONS[OS_VERSIONS.length - 1];
+
+  /** The supported DSE versions */
+  static final ProtocolVersion[] DSE_VERSIONS = new ProtocolVersion[] {DSE_V1, DSE_V2};
 
-  static final ProtocolVersion MIN_SUPPORTED_VERSION = SUPPORTED_VERSIONS[0];
-  static final ProtocolVersion MAX_SUPPORTED_VERSION =
-      SUPPORTED_VERSIONS[SUPPORTED_VERSIONS.length - 1];
+  static final ProtocolVersion MIN_DSE_VERSION = DSE_VERSIONS[0];
+  static final ProtocolVersion MAX_DSE_VERSION = DSE_VERSIONS[DSE_VERSIONS.length - 1];
+  private static boolean supportDseProtocol =
+      "DsePersistence".equals(System.getProperty("stargate.persistence_id"))
+          || "CndbPersistence".equals(System.getProperty("stargate.persistence_id"));
 
-  /** All supported versions, published as an enumset */
+  /** All supported versions */
   public static final EnumSet<ProtocolVersion> SUPPORTED =
-      EnumSet.copyOf(Arrays.asList(SUPPORTED_VERSIONS));
+      supportDseProtocol ? EnumSet.of(V3, V4, V5, DSE_V1, DSE_V2) : EnumSet.of(V3, V4, V5);
+
+  public static final List<String> SUPPORTED_VERSION_NAMES =
+      SUPPORTED.stream().map(ProtocolVersion::toString).collect(Collectors.toList());
 
   /** Old unsupported versions, this is OK as long as we never add newer unsupported versions */
   public static final EnumSet<ProtocolVersion> UNSUPPORTED = EnumSet.complementOf(SUPPORTED);
 
   /** The preferred versions */
-  public static final ProtocolVersion CURRENT = V4;
+  public static final ProtocolVersion CURRENT = supportDseProtocol ? DSE_V2 : V4;
```

Comment:
maybe not pertinent for right now, but why wouldn't preferred version be v5?

Reply:
V5 is still beta for DSE 6.8.